unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -159,6 +162,77 @@ public UTF8String toLowerCase() {
     return UTF8String.fromString(toString().toLowerCase());
   }
 
+  public UTF8String trim() {
+    return UTF8String.fromString(toString().trim());
```

Comment:
I am not sure if the if we can operate the bytes for UTF8, particularly for the suffix bytes for `trim` or `rightTrim`. as it may occurpt the string. 
For example, in UTF-8, a Chinese character is 3 bytes, and the last byte probably be 0x20 (space), however, we can simply cut off it.